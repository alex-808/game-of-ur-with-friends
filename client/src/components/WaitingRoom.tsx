import React from 'react'

interface Props {
  roomID: string
}

const WaitingRoom: React.FC<Props> = ({ roomID, children }) => {
  return (
    <div data-testid="waiting-room">
      {children}
      <div className="waiting-header">Waiting for partner</div>
      <span className="waiting-room">
        <div>Game code:</div>
        <span data-testid="roomID" className="roomID">
          {roomID}
        </span>
      </span>
    </div>
  )
}

export { WaitingRoom }
export type { Props }
