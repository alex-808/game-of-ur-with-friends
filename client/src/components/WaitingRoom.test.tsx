import React from 'react'
import {
  render,
  screen,
  fireEvent,
  waitForElement,
  cleanup,
} from '@testing-library/react'

import renderer from 'react-test-renderer'

import { WaitingRoom, Props } from './WaitingRoom'

function renderWaitingRoom(props: Partial<Props> = {}) {
  const defaultProps: Props = {
    roomID: '12345',
  }
  return render(<WaitingRoom {...defaultProps} {...props} />)
}

afterEach(cleanup)

describe('<WaitingRoom />', () => {
  it('display game code', async () => {
    const { findByTestId } = renderWaitingRoom()
    const waitingRoom = await findByTestId('waiting-room')

    expect(waitingRoom).toHaveTextContent('12345')
  })
  it('matches snapshot', async () => {
    const tree = renderer.create(<WaitingRoom roomID="12345" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
