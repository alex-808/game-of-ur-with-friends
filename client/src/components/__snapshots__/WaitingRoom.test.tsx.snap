// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WaitingRoom /> matches snapshot 1`] = `
<div
  data-testid="waiting-room"
>
  <div
    className="waiting-header"
  >
    Waiting for partner
  </div>
  <span
    className="waiting-room"
  >
    <div>
      Game code:
    </div>
    <span
      className="roomID"
      data-testid="roomID"
    >
      12345
    </span>
  </span>
</div>
`;
